src/client/api/pool.c

```diff
@@ -11,14 +11,18 @@
 #include "client_internal.h"
 #include "task_internal.h"
 
+/** Disable backward compat code */
+#undef daos_pool_connect
+
+/** Kept for backward ABI compatibility, but not advertised via header file */
 int
-daos_pool_connect(const uuid_t uuid, const char *grp,
-		  unsigned int flags,
+daos_pool_connect(const char *pool, const char *grp, unsigned int flags,
```

Comment:
Yes, MPI-IO driver and HDF5 haven't been recompiled and worked in the CI.
I had to do this because of daos_pool.h get indirectly included and this is causing a compile error due to function signature mismatch.

Glad that it worked with your testing :)